{
	"id": "oval:com.redhat.cve:def:202240897",
	"version": "636",
	"class": "vulnerability",
	"metadata": {
		"title": "pypa-setuptools: Regular Expression Denial of Service (ReDoS) in package_index.py (moderate)",
		"reference": [
			{
				"ref_id": "CVE-2022-40897",
				"ref_url": "https://access.redhat.com/security/cve/CVE-2022-40897",
				"source": "CVE"
			}
		],
		"description": "DOCUMENTATION: A flaw was found in Python Setuptools due to a regular expression Denial of Service (ReDoS) present in package_index.py. This issue could allow a remote attacker to cause a denial of service via HTML in a crafted package or custom PackageIndex page.",
		"advisory": {
			"from": "secalert@redhat.com",
			"severity": "Moderate",
			"updated": {
				"date": "2024-03-27"
			},
			"cve": [
				{
					"text": "CVE-2022-40897",
					"cvss3": "5.9/CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
					"cwe": "CWE-185",
					"href": "https://access.redhat.com/security/cve/CVE-2022-40897",
					"impact": "moderate",
					"public": "20221222"
				}
			],
			"affected": {
				"resolution": [
					{
						"state": "Out of support scope",
						"component": [
							"python-setuptools"
						]
					}
				]
			},
			"affected_cpe_list": {
				"cpe": [
					"cpe:/a:redhat:rhel_extras:6",
					"cpe:/a:redhat:rhel_extras_hpn:6",
					"cpe:/a:redhat:rhel_extras_oracle_java:6",
					"cpe:/a:redhat:rhel_extras_sap:6",
					"cpe:/a:redhat:rhel_extras_sap_hana:6",
					"cpe:/o:redhat:enterprise_linux:6",
					"cpe:/o:redhat:enterprise_linux:6::client",
					"cpe:/o:redhat:enterprise_linux:6::computenode",
					"cpe:/o:redhat:enterprise_linux:6::server",
					"cpe:/o:redhat:enterprise_linux:6::workstation"
				]
			},
			"issued": {}
		},
		"affected": {}
	},
	"criteria": {
		"operator": "OR",
		"criterias": [
			{
				"operator": "AND",
				"criterions": [
					{
						"test_ref": "oval:com.redhat.cve:tst:20022439069",
						"comment": "Red Hat Enterprise Linux 6 is installed"
					},
					{
						"test_ref": "oval:com.redhat.cve:tst:20131633001",
						"comment": "python-setuptools is installed"
					},
					{
						"test_ref": "oval:com.redhat.cve:tst:20131633002",
						"comment": "python-setuptools is signed with Red Hat redhatrelease2 key"
					}
				]
			}
		],
		"criterions": [
			{
				"test_ref": "oval:com.redhat.cve:tst:20022439070",
				"comment": "Red Hat Enterprise Linux must be installed"
			}
		]
	}
}
